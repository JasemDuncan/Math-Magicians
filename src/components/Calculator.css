*,
*::before,
*::after {
  box-sizing: border-box;
}

body {
  background: rgb(235, 236, 240);
  font-family: 'Times New Roman', Times, serif;
}

.App {
  width: 100%;
}

.calculator-grid {
  display: grid;
  justify-content: center;
  align-content: center;
  grid-template-columns: repeat(4, 100px);
  grid-template-rows: minmax(60px, auto) repeat(5, 60px);
}

.calculator-grid > button {
  cursor: pointer;
  font-size: 0.7rem;
  font-weight: bold;
  color: black;
  border: 1px solid rgb(207, 207, 207);
  outline: none;
}

.calculator-grid > button:hover {
  background-color: #fafafa;
}

.span-two {
  grid-column: span 2;
}

.output {
  grid-column: 1 / -1;
  background-color: rgb(123, 122, 135);
  display: flex;
  align-items: flex-end;
  flex-direction: column;
  padding: 10px;
  word-wrap: break-word;
  margin: 0;
}

.output .current-operand {
  color: white;
  font-size: 1.3rem;
}

.orange-buttton {
  background-color: rgb(227, 142, 64);
}
