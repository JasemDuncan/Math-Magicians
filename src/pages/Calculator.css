.contentCalculator {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
}

.title {
  display: flex;
  flex-direction: column;
  width: 40%;
  justify-content: center;
  align-items: center;
}

h1 {
  font-size: 2.8rem;
}

.calculator {
  display: flex;
  flex-direction: column;
  width: 60%;
  margin: 0%;
}
