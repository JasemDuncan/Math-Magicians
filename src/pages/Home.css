.content {
  display: flex;
  flex-direction: column;
}

h1 {
  display: flex;
  flex-direction: row;
  padding-left: 3%;
}

h2 {
  display: flex;
  flex-direction: row;
  padding-left: 3%;
}

.parragraph {
  display: flex;
  flex-direction: column;
  text-align: justify;
  padding: 0 3% 0 3%;
}
