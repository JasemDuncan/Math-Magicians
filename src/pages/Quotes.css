.content {
  display: flex;
  flex-direction: column;
  width: 100%;
  padding: 0 3% 0 3%;
}

.nameQuote {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-content: center;
  width: 100%;
  padding-top: 5%;
  font-size: 3.5rem;
}

.quote {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-content: center;
  width: 100%;
  padding-top: 5%;
  font-size: 1.7rem;
}
