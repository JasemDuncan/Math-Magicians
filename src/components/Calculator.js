import React from 'react';
import './Calculator.css';
import calculate from '../logic/calculate';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: 0,
      next: 0,
    };

    this.click = this.click.bind(this);
  }

  click(event) {
    this.setState((obj) => calculate(obj, event.target.innerText));
  }

  render() {
    let lblValue = 0;
    const { total, next } = this.state;
    if (next) {
      lblValue = next;
    } else if (total) {
      lblValue = total;
    }

    return (
      <div className="App">
        <div className="calculator-grid">
          <div className="output">
            <div className="current-operand">{lblValue}</div>
          </div>
          <button type="button" onClick={this.click}>AC</button>
          <button type="button" onClick={this.click}>+/-</button>
          <button type="button" onClick={this.click}>%</button>
          <button type="button" onClick={this.click} className="orange-buttton">/</button>
          <button type="button" onClick={this.click}>7</button>
          <button type="button" onClick={this.click}>8</button>
          <button type="button" onClick={this.click}>9</button>
          <button type="button" onClick={this.click} className="orange-buttton">x</button>
          <button type="button" onClick={this.click}>4</button>
          <button type="button" onClick={this.click}>5</button>
          <button type="button" onClick={this.click}>6</button>
          <button type="button" onClick={this.click} className="orange-buttton">-</button>
          <button type="button" onClick={this.click}>1</button>
          <button type="button" onClick={this.click}>2</button>
          <button type="button" onClick={this.click}>3</button>
          <button type="button" onClick={this.click} className="orange-buttton">+</button>
          <button type="button" onClick={this.click} className="span-two">0</button>
          <button type="button" onClick={this.click}>.</button>
          <button type="button" onClick={this.click} className="orange-buttton">=</button>
        </div>
      </div>
    );
  }
}

// const Calculator = () => (
//   <div className="App">
//     <div className="calculator-grid">
//       <div className="output">
//         <div className="current-operand">0</div>
//       </div>
//       <button type="button">AC</button>
//       <button type="button">+/-</button>
//       <button type="button">%</button>
//       <button type="button" className="orange-buttton">/</button>
//       <button type="button">7</button>
//       <button type="button">8</button>
//       <button type="button">9</button>
//       <button type="button" className="orange-buttton">x</button>
//       <button type="button">4</button>
//       <button type="button">5</button>
//       <button type="button">6</button>
//       <button type="button" className="orange-buttton">-</button>
//       <button type="button">1</button>
//       <button type="button">2</button>
//       <button type="button">3</button>
//       <button type="button" className="orange-buttton">+</button>
//       <button type="button" className="span-two">0</button>
//       <button type="button">.</button>
//       <button type="button" className="orange-buttton">=</button>
//     </div>
//   </div>
// );

export default Calculator;
