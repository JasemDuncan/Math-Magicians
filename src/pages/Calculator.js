import React from 'react';
import './Calculator.css';
import Calc from '../components/Calculator';

const Calculator = () => {
  const calculatorData = {
    title: 'Calculator!',
    subTitle: 'Lets do some math!',
  };

  const Page = () => {
    const { title, subTitle } = calculatorData;
    return (
      <>
        <div className="contentCalculator">
          <div className="title">
            <h1>{title}</h1>
            <h2>{subTitle}</h2>
          </div>
          <div className="calculator">
            <Calc />
          </div>
        </div>
      </>
    );
  };
  return (
    <Page />
  );
};

export default Calculator;
