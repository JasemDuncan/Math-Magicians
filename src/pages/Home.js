import React from 'react';
import './Home.css';

const Home = () => {
  const homeData = {
    title: 'Home',
    subHeading: 'Welcome!',
    description: (
      <>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed cursus congue
          velit, ut bibendum mauris cursus a.
          Integer risus nisl, iaculis vel sodales sed, sollicitudin scelerisque magna.
          Phasellus eu volutpat justo, in placerat purus. Suspendisse et leo quis
          ligula tincidunt sollicitudin. Quisque pulvinar massa pharetra est semper g
          ravida. Vestibulum turpis nulla, fringilla quis lacinia non, venenatis
          a massa. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras sit
          amet tincidunt est.
          Nullam et malesuada ipsum, ac elementum metus. Aenean fermentum porttitor
          quam volutpat gravida. Etiam sed libero a lacus dignissim convallis. Donec
          facilisis nibh id velit egestas posuere. In efficitur aliquet tincidunt.
          Sed ut posuere eros, vel dictum lectus. Mauris nec lobortis lectus. Phasellus
          tortor turpis, pretium sit amet iaculis a, cursus et turpis.

        </p>
        <p>
          Phasellus posuere eros vel nulla lacinia, eu sodales orci faucibus.
          Etiam vehicula, velit maximus tempus ultrices,
          massa ipsum fermentum odio, mattis consequat lectus odio vel nulla.
          Mauris egestas tincidunt lacus dapibus luctus.
          Cras lobortis malesuada dui, quis lacinia est efficitur non. Donec sit
          amet velit sit amet odio dapibus ultrices. Nunc condimentum
          ut arcu auctor tincidunt. Nam eu enim venenatis, fringilla augue vitae,
          sagittis ante. Sed et eleifend nunc. Quisque a
          arcu in lacus pharetra viverra ac vitae lectus. Nam aliquam, turpis sit
          amet placerat ullamcorper, arcu dolor convallis dui, ut mollis lorem mauris eget
          lorem. Maecenas ligula ligula, imperdiet sit amet laoreet sollicitudin,
          tempor sed lectus. Phasellus nec maximus elit, sed dictum erat.
        </p>
      </>
    ),
  };

  const Page = () => {
    const { title, subHeading, description } = homeData;
    return (
      <>
        <div className="content">
          <h1>{title}</h1>
          <h2>{subHeading}</h2>
          <div className="parragraph">{description}</div>
        </div>
      </>
    );
  };
  return (
    <Page />
  );
};

export default Home;
