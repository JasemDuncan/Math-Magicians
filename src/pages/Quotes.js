import React from 'react';
import './Quotes.css';

const Quotes = () => {
  const QuotesData = {
    name: 'William Paul Thurston',
    quote:
        'Mathematics is not about numbers, equations, computations, or algorithms: It is about understanding',
  };

  const Quote = () => {
    const { name, quote } = QuotesData;
    return (
      <>
        <div className="content">
          <div className="nameQuote">
            {name}
          </div>
          <div className="quote">
            {quote}
          </div>
        </div>
      </>
    );
  };

  return (
    <Quote />
  );
};

export default Quotes;
