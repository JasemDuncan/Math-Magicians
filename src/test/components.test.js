import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import Calc from '../components/Calculator';
import Calculator from '../pages/Calculator';
import Home from '../pages/Home';
import Quotes from '../pages/Quotes';

describe('', () => {
  it('Home renders correctly', () => {
    const tree = renderer
      .create(<Home />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Calculator renders correctly', () => {
    const tree = renderer
      .create(<Calculator />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Quotes renders correctly', () => {
    const tree = renderer
      .create(<Quotes />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Calc renders correctly', () => {
    const tree = renderer
      .create(<Calc />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
