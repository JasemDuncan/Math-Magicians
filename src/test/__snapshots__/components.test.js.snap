// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Calc renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="calculator-grid"
  >
    <div
      className="output"
    >
      <div
        className="current-operand"
      >
        0
      </div>
    </div>
    <button
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="orange-buttton"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="orange-buttton"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="orange-buttton"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="orange-buttton"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="span-two"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="orange-buttton"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;

exports[` Calculator renders correctly 1`] = `
<div
  className="contentCalculator"
>
  <div
    className="title"
  >
    <h1>
      Calculator!
    </h1>
    <h2>
      Lets do some math!
    </h2>
  </div>
  <div
    className="calculator"
  >
    <div
      className="App"
    >
      <div
        className="calculator-grid"
      >
        <div
          className="output"
        >
          <div
            className="current-operand"
          >
            0
          </div>
        </div>
        <button
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="orange-buttton"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="orange-buttton"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="orange-buttton"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="orange-buttton"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
        <button
          className="span-two"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="orange-buttton"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[` Home renders correctly 1`] = `
<div
  className="content"
>
  <h1>
    Home
  </h1>
  <h2>
    Welcome!
  </h2>
  <div
    className="parragraph"
  >
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed cursus congue velit, ut bibendum mauris cursus a. Integer risus nisl, iaculis vel sodales sed, sollicitudin scelerisque magna. Phasellus eu volutpat justo, in placerat purus. Suspendisse et leo quis ligula tincidunt sollicitudin. Quisque pulvinar massa pharetra est semper g ravida. Vestibulum turpis nulla, fringilla quis lacinia non, venenatis a massa. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras sit amet tincidunt est. Nullam et malesuada ipsum, ac elementum metus. Aenean fermentum porttitor quam volutpat gravida. Etiam sed libero a lacus dignissim convallis. Donec facilisis nibh id velit egestas posuere. In efficitur aliquet tincidunt. Sed ut posuere eros, vel dictum lectus. Mauris nec lobortis lectus. Phasellus tortor turpis, pretium sit amet iaculis a, cursus et turpis.
    </p>
    <p>
      Phasellus posuere eros vel nulla lacinia, eu sodales orci faucibus. Etiam vehicula, velit maximus tempus ultrices, massa ipsum fermentum odio, mattis consequat lectus odio vel nulla. Mauris egestas tincidunt lacus dapibus luctus. Cras lobortis malesuada dui, quis lacinia est efficitur non. Donec sit amet velit sit amet odio dapibus ultrices. Nunc condimentum ut arcu auctor tincidunt. Nam eu enim venenatis, fringilla augue vitae, sagittis ante. Sed et eleifend nunc. Quisque a arcu in lacus pharetra viverra ac vitae lectus. Nam aliquam, turpis sit amet placerat ullamcorper, arcu dolor convallis dui, ut mollis lorem mauris eget lorem. Maecenas ligula ligula, imperdiet sit amet laoreet sollicitudin, tempor sed lectus. Phasellus nec maximus elit, sed dictum erat.
    </p>
  </div>
</div>
`;

exports[` Quotes renders correctly 1`] = `
<div
  className="content"
>
  <div
    className="nameQuote"
  >
    William Paul Thurston
  </div>
  <div
    className="quote"
  >
    Mathematics is not about numbers, equations, computations, or algorithms: It is about understanding
  </div>
</div>
`;
